client.go

```diff
@@ -146,16 +146,16 @@ func (c *Client) GetMap(ctx context.Context, name string) (*Map, error) {
 	if c.ic.State() != client.Ready {
 		return nil, hzerrors.ErrClientNotActive
 	}
-	m, err := c.proxyManager.getMap(ctx, name)
-	if err != nil {
-		return nil, err
-	}
-	ncc, ok, err := c.cfg.GetNearCache(name)
-	if err != nil {
-		return nil, err
-	}
-	if ok {
-		// there is a near cache config for this map
+	return c.proxyManager.getMap(ctx, name, func(p *proxy) (interface{}, error) {
+		m := newMap(p)
+		ncc, ok, err := c.cfg.GetNearCache(name)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			// there is no near cache config for this map
+			return m, nil
+		}
```

Comment:
I did not really get what is different and fixing the issue here.

Reply:
A new backing map for the near cache is created every time `GetMap` is called, this PR fixes that.